### PR TITLE
Create cmdlineeditors.md

### PR DESCRIPTION
**Pull request by andeElliott**

### CODE REVIEW

File: chapter1/cmdlineeditors.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+This course is not designed to teach you to use all of these tools however if
+you are interested please see ABC for further details.
+###[/supplemental]
+
+The default text editor on many systems is `vim` an advanced editor with a
+very steep learning curve, if you are interested in learning vim we would
+recommend having a look at ABCDEF.
+
+However, for this course, to keep this simple we will use the `nano` text
+editor - a simple editor in which you cansimpyl directly input text. To fix
```
Comment (samcolvi) [inline chapter1/cmdlineeditors.md:28]: can simply*

File: chapter1/cmdlineeditors.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+### Text Editors
+
+To use git on the command line it can be helpful (although not required) may
```
Comment (samcolvi) [inline chapter1/cmdlineeditors.md:3]: take out 'may'

File: chapter1/cmdlineeditors.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+to edit text files on the command line, this can help with commit messages or
+just making quick edits to files.
+
+It is important to at least somewhat know how to do this, as otherwise you may
+get stuck e.g. if you do not specify a message for some commands it will open a
```
Comment (samcolvi) [inline chapter1/cmdlineeditors.md:8]: maybe describe what being 'stuck' actually is

File: chapter1/cmdlineeditors.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+```{bash}
+git config --global core.editor "nano"
+```
+
+After this command the default editor will be nano.
+
+###[supplemental]
+
+Getting stuck in `vim`.
```
Comment (samcolvi) [inline chapter1/cmdlineeditors.md:39]: is it worth having this as a small heading?

File: chapter1/cmdlineeditors.md (hunk 1 above)
Comment (andeElliott) [reply chapter1/cmdlineeditors.md:28]: Done

File: chapter1/cmdlineeditors.md (hunk 3 above)
Comment (andeElliott) [reply chapter1/cmdlineeditors.md:8]: Done

File: chapter1/cmdlineeditors.md (hunk 4 above)
Comment (andeElliott) [reply chapter1/cmdlineeditors.md]: restructured the section now, but happy to add if you think it is still worth it?

**Comment by andeElliott** [comment]
> Also suggest adding TextMate as a code editor with a GUI

@dmca-glasgow Sorry, dont know how to do that (I assume you pass a path to the exec?) but happy for you to add?

**Comment by dmca-glasgow** [comment]
> > Also suggest adding TextMate as a code editor with a GUI
> 
> @dmca-glasgow Sorry, dont know how to do that (I assume you pass a path to the exec?) but happy for you to add?

I'm not sure how to do it either, I usually use Source Tree. I can look it up when I'm back from annual leave if you're having issues.

**Pull request merged**